### PR TITLE
Add support for authentication to Java time client

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Moves #3626 forward.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/TimeClient.java [hunk 1]
```diff
@@ -12,6 +12,8 @@
 public interface TimeClient {
 
     Single<Empty> setTime(Instant currentTime, Instant newTime);
+    Single<Empty> setTime(Instant currentTime, Instant newTime, String accessToken);
```
Comment (miklos-da) [inline language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/TimeClient.java]: Are we duplicating the interface methods for backwards compatibility? We may want to simply add a method with `Optional` access token and deprecate the old one.

Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/TimeClient.java]: I'd prefer keeping backwards compatibility unless strictly needed, but thanks for the feedback.

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/TimeClientImplTest.scala [hunk 2]
```diff
@@ -71,14 +83,73 @@ class TimeClientImplTest extends FlatSpec with Matchers with OptionValues with D
...
+        }
+      }
+    }
+  }
+
+  it should "deny access with insufficient authorization" in {
+    withClue("getTime") {
+      expectPermissionDenied {
+        toAuthenticatedServer {
+          _.getTime(emptyToken).blockingFirst()
```
Comment (rautenrieth-da) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/TimeClientImplTest.scala]:
```suggestion
          _.getTime(somePartyReadWriteToken).blockingFirst()
```
Using `emptyToken` is better for checking whether empty tokens are handled correctly, using `somePartyReadWriteToken` checks whether this endpoint is admin-only (which it should be).

Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/TimeClientImplTest.scala]: `getTime` is public :wink: 
```scala
  override def getTime(
      request: GetTimeRequest,
      responseObserver: StreamObserver[GetTimeResponse]): Unit =
    authorizer.requirePublicClaimsOnStream(service.getTime)(request, responseObserver)
```

Comment (rautenrieth-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/TimeClientImplTest.scala]: Disregard the above, the `getTime` call requires public claims only. The information at https://github.com/digital-asset/daml/blob/master/ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala#L71 should be updated.

Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/TimeClientImplTest.scala]: I've addressed it in this PR https://github.com/digital-asset/daml/pull/3727

**Pull request merged**